subprojects/core/src/main/java/org/gradle/api/internal/tasks/NonCacheableTaskOutputPropertySpec.java

```diff
@@ -55,7 +61,7 @@ public PathNormalizationStrategy getPathNormalizationStrategy() {
 
     @Override
     public int compareTo(TaskPropertySpec o) {
-        return parent.compareTo(o);
+        return getPropertyName().compareTo(o.getPropertyName());
```

Comment:
Wow, nice catch. Was this also causing problems?

Reply:
It was not causing problems, I just saw the properties in a "wrong" order when test test did output them.

Reply:
Ah, I saw the out of order property list, too, but put off investigating. Good job tracking that down!